### ISSUE TITLE
Optimization Level configuration

### ISSUE THREAD

**Issue by Mazyod** [ReactiveCocoa/ReactiveSwift#552]
I've been scratching my head all day as to why our new Carthage static libs setup works great for all our dependencies except Reactive(ObjC|Swift|Cocoa). For these frameworks, when I try to perform `-all_load` to link the static frameworks against one monolithic dynamic framework, I kept getting these errors when building for Simulator:

```
Undefined symbols for architecture x86_64:
  "___llvm_profile_runtime", referenced from:
      ___llvm_profile_runtime_user in ReactiveSwift(Signal.o)
      ___llvm_profile_runtime_user in ReactiveSwift(Optional.o)
      ___llvm_profile_runtime_user in ReactiveSwift(Disposable.o)
      ___llvm_profile_runtime_user in ReactiveSwift(SignalProducer.o)
      ___llvm_profile_runtime_user in ReactiveSwift(ResultExtensions.o)
      ___llvm_profile_runtime_user in ReactiveSwift(ValidatingProperty.o)
      ___llvm_profile_runtime_user in ReactiveSwift(Deprecations+Removals.o)
      ...
     (maybe you meant: ___llvm_profile_runtime_user)
ld: symbol(s) not found for architecture x86_64
```

One workaround I found was to simply add `-fprofile-instr-generate` to other linker flags, but looking into why is this required in the first place lead me to find out that optimization level for Release configuration on the mentioned frameworks is set to `-Onone`. I tried searching git history and github for the reason to set this, alas I couldn't find a reason.

I tried to override the setting in our custom carthage xcconfig file with `GCC_OPTIMIZATION_LEVEL = s`, but that didn't yield any results.

**Comment by andersio** [comment]
I think it was unintentional, since we do not have specific reason to customise it, where the inherited configuration from `jspahrsummers/xcconfigs` has it enabled.

**Comment by andersio** [comment]
As for needing `-fprofile-instr-generate`, IIRC we disabled code coverage because of App Store submission issue. Though apparently Carthage 0.26 has formally addressed the issue. Perhaps we can now reenable it.

**Comment by andersio** [comment]
Hi, please check 3.0 which should have addressed both issues raised here (#553).

**Comment by mfclarke** [comment]
Unfortunately this issue persists with Xcode 9.2. The cause now is code coverage being enabled on the framework scheme's test action 😮

A workaround is to create and share a dedicated tests scheme with coverage enabled, so the framework scheme can have this option disabled.

Though a real solution might be to somehow override this setting in the scheme while running libtool. More of an issue for the Carthage repo perhaps?
